cvmfs/cvmfs_server

```diff
@@ -4046,8 +4052,20 @@ snapshot() {
       -i ${spool_dir}/tmp/last_snapshot \
       -o .cvmfs_last_snapshot"
 
-    # remove magic file after successful replication
-    $user_shell "$delete_command"
+    # run the automatic garbage collection (if configured)
+    if is_garbage_collectable $name && [ ! -z "$CVMFS_GC_TIMESTAMP_THRESHOLD" ]
```

Comment:
> In the long-term we might have the cvmfs_server gc as a public command, to explicitly run the garbage collection instead of running it automatically after each publish.  With CVMFS_GARBAGE_COLLECTION=true the basic functionality of garbage collection is switched on. 

Couldn't/shouldn't the functionality of garbage collection be turned on by looking for 'Gyes' in the .cvmfspublished file?  Or are you saying that if 'cvmfs_server gc' became a public function you wouldn't require that but instead go by the environment variable?  It seems to me that if somebody ran the gc function by hand, it ought to be done whether or not an environment variable setting was turned on.

> CVMFS_GC_TIMESTAMP_THRESHOLD is meant to additionally enable the automatic garbage collection on publish/snapshot. Though, I agree that this is not obvious and propose to rename the CVMFS_GC_TIMESTAMP_THRESHOLD to CVMFS_AUTO_GC_TIMESPAN. Does that seem more clear to you?

That's a bit of an improvement but it still kind of implies to me that it's the timespan for auto gc and not that it enables auto gc.  I would rather replace CVMFS_GARBAGE_COLLECTION with CVMFS_AUTO_GARBAGE_COLLECTION to enable automatic garbage collection, and have a default for the threshold.  I don't see why most people should care about the threshold, a default of 3 days will be fine.